staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/admission.go

```diff
@@ -299,7 +300,7 @@ func (a *ValidatingAdmissionWebhook) shouldCallHook(h *v1beta1.Webhook, attr adm
 	return a.namespaceMatcher.MatchNamespaceSelector(h, attr)
 }
 
-func (a *ValidatingAdmissionWebhook) callHook(ctx context.Context, h *v1beta1.Webhook, attr admission.Attributes) error {
+func (a *ValidatingAdmissionWebhook) callHook(ctx context.Context, h *v1beta1.Webhook, attr admission.Attributes, annotationLock sync.Mutex) error {
```

Comment:
this is ugly

Reply:
What about we move this mutex into SetAnnotations()?

Reply:
yes. And get rid of the single k/v SetAnnotations variant, but only for a whole map.

Reply:
Done.